crates/e2e/tests/e2e/protocol_fee.rs

```diff
@@ -215,9 +214,10 @@ async fn combined_protocol_fees(web3: Web3) {
         .await;
 
     tracing::info!("Waiting for liquidity state to update");
+    // Use a more robust approach to wait for liquidity state updates
+    // Instead of manually mining blocks and checking for 2x quote changes,
+    // we'll wait for the quotes to stabilize after the pool changes
     wait_for_condition(TIMEOUT, || async {
-        // Mint blocks until we evict the cached liquidity and fetch the new state.
-        onchain.mint_block().await;
```

Comment:
I think the block mining is needed here (or somewhere at least). The reason is that liquidity states can be cached for faster quoting and the cached state will only be evicted when the state is sufficiently old (compared to the current block number).